### PR TITLE
IGNITE-14350 Added distributed property to disable CDC.

### PR DESCRIPTION
**Pull request by NSAmelchev**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java [hunk 1]
```diff
@@ -409,6 +416,9 @@ public class FileWriteAheadLogManager extends GridCacheSharedManagerAdapter impl
     /** Pointer to the last successful checkpoint until which WAL segments can be safely deleted. */
     private volatile WALPointer lastCheckpointPtr = new WALPointer(0, 0, 0);
 
+    /** Force disable CDC. */
+    private final DistributedBooleanProperty cdcForceDisable = detachedBooleanProperty(CDC_FORCE_DISABLE);
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java:420]: Let's rename to `cdcDisabled`

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java [hunk 2]
```diff
@@ -220,6 +224,9 @@ public class FileWriteAheadLogManager extends GridCacheSharedManagerAdapter impl
     /** @see IgniteSystemProperties#IGNITE_THRESHOLD_WAIT_TIME_NEXT_WAL_SEGMENT */
     public static final long DFLT_THRESHOLD_WAIT_TIME_NEXT_WAL_SEGMENT = 1000L;
 
+    /** CDC force disable distributed property name. */
+    public static final String CDC_FORCE_DISABLE = "cdc.forceDisable";
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java:228]: Let's rename to `cdc.disabled`

File: modules/core/src/main/java/org/apache/ignite/internal/cluster/DistributedConfigurationUtils.java [hunk 3]
```diff
@@ -33,8 +33,10 @@
  * Distributed configuration utilities methods.
  */
 public final class DistributedConfigurationUtils {
-    /**
-     */
+    /** Distributed property update message. */
+    public static final String PROPERTY_UPDATE_MSG = "Distributed property '%s' was changed from '%s' to '%s'.";
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/cluster/DistributedConfigurationUtils.java:37]: One usage. Let's inline.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java [hunk 4]
```diff
@@ -2118,7 +2141,7 @@ public SegmentArchiveResult archiveSegment(long absIdx) throws StorageException
 
                 Files.move(dstTmpFile.toPath(), dstFile.toPath());
 
-                if (walCdcDir != null)
+                if (walCdcDir != null && !cdcForceDisable.getOrDefault(false))
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java:2144]: Let's rewrite a bit and warn user every time we skip link creation.

```
if (walCdcDir != null) {
    if (!cdcForceDisable.getOrDefault(false))
        Files.createLink(walCdcDir.toPath().resolve(dstFile.getName()), **dstFile.toPath());
    else 
        log.warning("Creation of segment CDC link skiped. `cdc.disabled` distributed property is false."); 
}
```

**Review by nizhikov — APPROVED** [review]

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java (hunk 4 above)
Comment (NSAmelchev) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/internal/cluster/DistributedConfigurationUtils.java (hunk 3 above)
Comment (NSAmelchev) [reply modules/core/src/main/java/org/apache/ignite/internal/cluster/DistributedConfigurationUtils.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java (hunk 2 above)
Comment (NSAmelchev) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java (hunk 1 above)
Comment (NSAmelchev) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java]: Done.

**Pull request merged**